src/features/live_sensor/atoms/collectedPoints.ts

```diff
@@ -0,0 +1,65 @@
+import { createAtom } from '~utils/atoms/createPrimitives';
+import { sensorDataAtom } from './sensorData';
+import { triggerRequestAction } from './triggerResource';
+import type { FeatureCollection } from '~utils/geoJSON/helpers';
+import type { UncertainNumber } from '../utils';
+
+const defaultState = { type: 'FeatureCollection', features: [] };
+
+export type CoordinatesData = {
+  lng: UncertainNumber;
+  lat: UncertainNumber;
+  alt: UncertainNumber;
+  altAccuracy: UncertainNumber;
+  speed: UncertainNumber;
+  accuracy: UncertainNumber;
+  heading: UncertainNumber;
+  coordTimestamp: number;
+  coordSystTimestamp: number;
+};
+
+// This atom responsible for collecting sensor data as geojson points entities
+// We need that due to our process of building analytical geospatial layers
+// It can either add a new feature formed from current sensor data or reset itself
+export const collectedPointsAtom = createAtom(
+  {
+    addFeature: (data: CoordinatesData) => data,
+    reset: () => null,
+  },
+  ({ onAction, getUnlistedState, schedule }, state: FeatureCollection = defaultState) => {
+    const sensorData = getUnlistedState(sensorDataAtom);
+
+    onAction('addFeature', (data) => {
+      if (!sensorData) return state;
```

Comment:
Do we need to add the feature `data` if exists to `state`?

Reply:
sorry didn't understand the question, can you be more verbose?